### PR TITLE
Standardise Templates 

### PR DESCRIPTION
**Pull request by mpguerra**
Major update to the nginx OAuth2 templates to:
- Standardise functions used across flows
- Follow same execution flow across flows
- Fix bugs with variable names and setup

### CODE REVIEW

**Comment by mpguerra** [comment]
closes #17 and #18

File: oauth2/client-credentials-flow/token-generation/nginx.conf [hunk 1]
```diff
@@ -35,7 +35,7 @@ http {
     set_by_lua $deployment 'return os.getenv("THREESCALE_DEPLOYMENT_ENV")';
     set $threescale_backend "https://su1.3scale.net:443";
 
-    location /_threescale/auth {
+    location = /_threescale/check_credentials {
```
Comment (mikz) [inline oauth2/client-credentials-flow/token-generation/nginx.conf]: Is this a typo?

Comment (mpguerra) [inline oauth2/client-credentials-flow/token-generation/nginx.conf]: nope, I wanted exact location matching in this case

Comment (mikz) [inline oauth2/client-credentials-flow/token-generation/nginx.conf]: Then maybe a comment would be nice. Also, why it is not for the others?

Comment (mikz) [inline oauth2/client-credentials-flow/token-generation/nginx.conf]: So ed39615 is something we should do for out generated configs too, right?

Comment (mpguerra) [inline oauth2/client-credentials-flow/token-generation/nginx.conf]: maybe... I don't want to dictate on the generated ones yet :) 

I have a looooot of suggestions :P

**Comment by mikz** [comment]
Looks like this reverted https://github.com/3scale/nginx-oauth-templates/pull/13

**Pull request merged**